.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "unit-test/background/reference-tests"]
+	path = unit-test/background/reference-tests
+	url = https://github.com/duckduckgo/privacy-reference-tests
```

Comment:
Can this by pinned to a specific tag or commit? I'd prefer updates to the tests to be explicit, otherwise a commit with passing tests could break in the future if the tests are updated.

Alternatively, could we publish the tests on npm (e.g. in the @duckduckgo namespace). This would allow us to pin the version, and newer versions of the tests would trigger dependabot to automatically open a pull request.

Reply:
This question was raised in the tech design and its definitely an interesting tradeoff.  There are a few intended mitigations:
- add a platform exception to the test definition.  This isn't yet supported the extension currently as its the reference implementation but looks like we'll have to have one for iOS at least at launch.  This is better than pinning because it makes it explicit which test is disabled for a given platform, and you still get the benefit of all of the other test being up to date.  My experience with pinning to a commit is that it tends to be a temporary stopgap that often never gets removed because its not obvious what the issue is that caused the pin in the first place.
- pinning to a commit is a fallback but we should not be changing ref tests without validating them across our products beforehand, and adding any unsupported platforms to the exception list before the new tests are merged.

It sounds like a more specific concern is around tests changing for features that are in progress.  I would argue there though that its the lesser of two evils.  If the tests are validated across main/master on each product, and a particular WIP branch goes red once they are merged to the test repo, I would rather that branch go red before being merged (vs after) as it has regressed something?